src/webgpu/util/texture/texel_data.ts

```diff
@@ -490,6 +490,17 @@ export const kTexelRepresentationInfo: {
       decode: applyEach(() => unreachable('depth24plus cannot be decoded'), [TexelComponent.Depth]),
       pack: () => unreachable('depth24plus data cannot be packed'),
     },
+    stencil8: {
+      componentOrder: [TexelComponent.Stencil],
+      encode: applyEach((n: number) => (assertInIntegerRange(n, 8, false), n), [
+        TexelComponent.Stencil,
+      ]),
+      decode: applyEach((n: number) => (assertInIntegerRange(n, 8, false), n), [
+        TexelComponent.Stencil,
+      ]),
+      componentInfo: { Stencil: { dataType: 'uint', bitLength: 8 } },
+      pack: components => packComponents([TexelComponent.Stencil], components, 8, 'uint'),
+    },
```

Comment:
nit: This is a little hard to read due to being over-general (not to say that other code here isn't hard to read, but); we can make it more readable by using a similar pattern to the one in depth24plus-stencil8 just below here. Rough sketch:

```ts
    'depth24plus-stencil8': {
      componentOrder: [TexelComponent.Stencil],
      componentInfo: {
        Stencil: { dataType: 'uint', bitLength: 8 },
      },
      encode: components => {
        assert(components.Stencil !== undefined);
        assertInIntegerRange(components.Stencil, 8, false);
        return components;
      },
      decode: components => {
        assert(components.Stencil !== undefined);
        assertInIntegerRange(components.Stencil, 8, false);
        return components;
      },
      pack: components => packComponents([TexelComponent.Stencil], components, 8, 'uint'),
    },
```